socket_wrapper.hpp

```diff
@@ -2,30 +2,32 @@
 #include "util.h"
```

Comment:
Lets get rid of this. This header (with a lot of other headers we don't really need) really significantly affects compilation time. And we're incuding it twice (here and in main file), which makes this even worse.
Better to use only headers you really need here, and in any other places.